service/storage_proxy.cc

```diff
@@ -5511,6 +5517,15 @@ class speculating_read_executor : public abstract_read_executor {
 public:
     using abstract_read_executor::abstract_read_executor;
     virtual void make_requests(digest_resolver_ptr resolver, storage_proxy::clock_type::time_point timeout) override {
+        slogger.debug("speculating_read_executor::make_requests targetrs {}", _targets);
```

Comment:
do we also want these logs in the other executor method? (perhaps you could make a common validation function for both?)

Reply:
Yes, it's unnecessary debug log. However regarding the check IMHO it does not make sense to introduce an additional dependency just because of the check. And also the logic and conditions could be changed the future.

Reply:
Removed in the last version.